tests/postgres_tests/test_json.py

```diff
@@ -152,6 +152,19 @@ def test_isnull_key(self):
             [self.objs[7], self.objs[8]]
         )
 
+    def test_none_key(self):
+        self.assertSequenceEqual(
+            JSONModel.objects.filter(field__j=None),
+            [self.objs[8]]
+        )
+
+    def test_none_key_exclude(self):
+        obj = JSONModel.objects.create(field={'j': 1})
+        self.assertSequenceEqual(
+            JSONModel.objects.exclude(field__j=None),
+            [obj]
+        )
```

Comment:
@Stranger6667 I was looking at this again. 

The thing that **always** comes up with `null` keys is distinguishing between the `not has_key()` case (which would be `field__j__is_null=True` here) and the case where the key exist but is `null`. 

People spend a **lot** of time being confused about the relationships between these two. 

SO for this `exclude` — `exclude(field__j=None)` — it may be natural to read that as 

> "Exclude(Has the key `j` and that key is `null`)" 

... whereas it's actually being interpreted as

> "Exclude(Either `not has_key('j')` or (Has the key `j` and that key is `null`))"

So question: Why is this the right behaviour? 

If it is the right behaviour we should bring it out in the docs, together with an example (and a test case) of how to achieve the other result. 

Does that make sense?

Reply:
@carltongibson 
sorry for not getting back to you, I'll look at it tomorrow

Reply:
Hey @Stranger6667. No problem. With the changes from https://github.com/Stranger6667/django/pull/1 (suitably tidied) my concern over this point is ≈resolved. 

For me, it's just a test on the new `Lookup` API in core and then a final nitpick.